LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift

```diff
@@ -0,0 +1,159 @@
+//
+//  FacebookLoginPlugin.swift.swift
+//  ZappLoginPlugins
+//
+//  Created by Roi Kedarya on 17/05/2018.
+//  Copyright © 2018 Applicaster. All rights reserved.
+//
+
+import Foundation
+import ZappPlugins
+import ApplicasterSDK
+import FacebookCore
+import FacebookLogin
+
+@objc class FacebookLoginPlugin: NSObject, ZPLoginProviderProtocol, ZPLoginProviderUserDataProtocol {
+    
+    public var configurationJSON: NSDictionary?
+    var loginCompletion: ((ZappPlugins.ZPLoginOperationStatus) -> Swift.Void)?
+    var fbToken: AccessToken?
+    var loginManager:LoginManager?
+    
+    public required override init() {
+        super.init()
+    }
+    
+    
+//MARK: ZPLoginProviderUserDataProtocol
+    
+    /**
+     
+    */
+    public func isUserComply(policies:[String: NSObject]) -> Bool {
+        return self.isAuthenticated()
+    }
+    
+    /**
+     Getter to the user Token usually can be used for authentication check
+    */
+    public func getUserToken() -> String {
+        var token = ""
+        if let userToken = UserDefaults.standard.string(forKey: "fb_login_token") {
+            token = userToken
+        }
+        return token
+    }
+    
+    /**
+     Setter for the user Token - usually set after login success
+     @Params: the authentication recieved when login successfuly
+     */
+    public func setUserToken(token: String?) {
+        UserDefaults.standard.set(token, forKey: "fb_login_token")
+    }
+    
+//MARK: ZPLoginProviderProtocol
+    
+    /**
+     Initialization of login Plugin instance
+     @Params: configurationJSON - dictionary containing the Plugin setitngs as defined in the plugin manifest
+     */
+    public required init(configurationJSON: NSDictionary?) {
+        super.init()
+        self.loginManager = LoginManager()
+        self.configurationJSON = configurationJSON
+    }
+    
+    /**
+     Test User in order to login successfuly use these credentials
+     user name: jorge_mtglsje_valdano@tfbnw.net
+     password: applicaster is the best
+    **/
+    
+    /**
+     This method is being called in order to start login process.
```

Comment:
can you explain in which cases this method is called? (in the sdk flow)

Reply:
I added an flow example